### PR TITLE
Expose publicly the different file extensions list to allow easier consumption of this information.

### PR DESCRIPTION
**Pull request by crousselle**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @crousselle has signed the CLA. Thanks for the pull request!

**Comment by crousselle** [comment]
@thuyBox @adempsey  a bit of context : in order to use the `file_extensions` parameter of our Search API endpoint, we need to provide a succession of string representing the file extensions we're searching for.
As in our app's UI, we'll be searching by file type (ie image, video, etc) and not file extensions, we need to have access to the data we're using to determine the type of a given file.
Moving it to a helper makes more sense, as there is no reason as to why it should be in a UIImage category.

File: BoxBrowseSDK/BoxBrowseSDK/Categories/UIImage+BOXBrowseSDKAdditions.h [hunk 1]
```diff
@@ -59,6 +59,14 @@ NS_ASSUME_NONNULL_BEGIN
  */
 + (UIImage *)box_genericFolderIcon;
 
+
++ (NSSet *)imageFileExtensions;
```
Comment (boxtcarpel) [inline BoxBrowseSDK/BoxBrowseSDK/Categories/UIImage+BOXBrowseSDKAdditions.h]: So... these should *not* be in a UIImage category, right?

Comment (crousselle) [inline BoxBrowseSDK/BoxBrowseSDK/Categories/UIImage+BOXBrowseSDKAdditions.h]: thanks for pointing that out, forgot to amend the fix. Done.

**Comment by thuyBox** [comment]
+2

File: BoxBrowseSDK/BoxBrowseSDK/Helpers/BOXBrowseSDKFileTypeHelper.m [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    
+    return extensions;
+}
+
++ (NSSet <NSString *> *)videoFileExtensions
+{
+    static NSSet *extensions = nil;
+    
+    if (extensions == nil) {
+        extensions = [NSSet setWithObjects:@"avi", @"flv", @"m4v", @"mov", @"mp4", @"mpeg", @"mpg", @"qt", @"wmv", nil];
```
Comment (boxtcarpel) [inline BoxBrowseSDK/BoxBrowseSDK/Helpers/BOXBrowseSDKFileTypeHelper.m]: Please include mts for video, it was previously missing.

File: BoxBrowseSDK/BoxBrowseSDK/Helpers/BOXBrowseSDKFileTypeHelper.m [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    
+    return extensions;
+}
+
++ (NSSet <NSString *> *)presentationFileExtensions
+{
+    static NSSet *extensions = nil;
+    
+    if (extensions == nil) {
+        extensions = [NSSet setWithObjects:@"gslide", @"key", @"opd", @"otp", @"pot", @"potx", @"ppt", @"pptx", nil];
```
Comment (boxtcarpel) [inline BoxBrowseSDK/BoxBrowseSDK/Helpers/BOXBrowseSDKFileTypeHelper.m]: please include 'keynote' in addition to 'key'.

File: BoxBrowseSDK/BoxBrowseSDK/Helpers/BOXBrowseSDKFileTypeHelper.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
++ (NSSet <NSString *> *)videoFileExtensions;
++ (NSSet <NSString *> *)docFileExtensions;
++ (NSSet <NSString *> *)codeFileExtensions;
++ (NSSet <NSString *> *)textFileExtensions;
++ (NSSet <NSString *> *)compressedFileExtensions;
++ (NSSet <NSString *> *)presentationFileExtensions;
++ (NSSet <NSString *> *)sheetFileExtensions;
++ (NSSet <NSString *> *)dbFileExtensions;
++ (NSSet <NSString *> *)iconFileExtensions;
+
```
Comment (boxtcarpel) [inline BoxBrowseSDK/BoxBrowseSDK/Helpers/BOXBrowseSDKFileTypeHelper.h]: What about box-dicom or box-notes? Would we treat them differently or handle them the same way?

**Comment by boxtcarpel** [comment]
Don't forget that the share SDK has an equivalent setup.

File: BoxBrowseSDK/BoxBrowseSDK/Helpers/BOXBrowseSDKFileTypeHelper.m (hunk 2 above)
Comment (crousselle) [inline BoxBrowseSDK/BoxBrowseSDK/Helpers/BOXBrowseSDKFileTypeHelper.m]: done

File: BoxBrowseSDK/BoxBrowseSDK/Helpers/BOXBrowseSDKFileTypeHelper.m (hunk 3 above)
Comment (crousselle) [inline BoxBrowseSDK/BoxBrowseSDK/Helpers/BOXBrowseSDKFileTypeHelper.m]: done

**Comment by crousselle** [comment]
@boxtcarpel added the missing extensions, and opened https://gitenterprise.inside-box.net/Mobile/box-ios-share-sdk-source/pull/123 for browse SDK.

**Pull request merged**